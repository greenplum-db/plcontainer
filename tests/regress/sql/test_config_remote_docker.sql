-- start_ignore
\! plcontainer runtime-add -r plc_python3_shared_remote_docker -i ${CONTAINER_NAME_SUFFIX_PYTHON} -l python3 -s enable_network=yes -s roles=$USER
\! plcontainer runtime-backup -f test_config_remote_docker.xml
\! ./data/test_assign_backend.py ./test_config_remote_docker.xml -r plc_python3_shared_remote_docker
\! plcontainer runtime-restore -f test_config_remote_docker.xml
\! rm test_config_remote_docker.xml
-- end_ignore
select * from plcontainer_refresh_config order by gp_segment_id DESC;

create function with_conf_remote_docker() returns TEXT as $$
  # container: plc_python3_shared_remote_docker
  return 'hello'
$$ LANGUAGE plcontainer;

select * from with_conf_remote_docker();

-- start_ignore
\! plcontainer runtime-delete -r plc_python3_shared_remote_docker
-- end_ignore

drop function with_conf_remote_docker();

