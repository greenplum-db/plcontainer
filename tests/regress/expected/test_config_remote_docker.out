-- start_ignore
\! plcontainer runtime-add -r plc_python3_shared_remote_docker -i ${CONTAINER_NAME_SUFFIX_PYTHON} -l python3 -s enable_network=yes -s roles=$USER
20230913:10:57:11:027585 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[WARNING]:-image python39.alpine is not in the list of 'docker images'
20230913:10:57:11:027585 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20230913:10:57:11:027585 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
\! plcontainer runtime-backup -f test_config_remote_docker.xml
\! ./data/test_assign_backend.py ./test_config_remote_docker.xml -r plc_python3_shared_remote_docker
\! plcontainer runtime-restore -f test_config_remote_docker.xml
20230913:10:57:13:028100 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[WARNING]:-image python39.alpine is not in the list of 'docker images'
20230913:10:57:13:028100 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20230913:10:57:13:028100 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
\! rm test_config_remote_docker.xml
-- end_ignore
select * from plcontainer_refresh_config order by gp_segment_id DESC;
 gp_segment_id | plcontainer_refresh_local_config 
---------------+----------------------------------
             2 | ok
             1 | ok
             0 | ok
            -1 | ok
(4 rows)

create function with_conf_remote_docker() returns TEXT as $$
  # container: plc_python3_shared_remote_docker
  return 'hello'
$$ LANGUAGE plcontainer;
select * from with_conf_remote_docker();
 with_conf_remote_docker 
-------------------------
 hello
(1 row)

-- start_ignore
\! plcontainer runtime-delete -r plc_python3_shared_remote_docker
20230913:10:57:18:028635 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Distributing file plcontainer_configuration.xml to all locations...
20230913:10:57:18:028635 plcontainer:01563e2c-e67b-4d29-5ed0-8a91b538dd15:gpadmin-[INFO]:-Configuration has changed. Run "select * from plcontainer_refresh_config" in open sessions. New sessions will get new configuration automatically.
-- end_ignore
drop function with_conf_remote_docker();
